app/src/lib/git/reset.ts

```diff
@@ -43,8 +43,8 @@ export async function reset(
   mode: GitResetMode,
   ref: string
 ): Promise<true> {
-  const modeFlag = resetModeToFlag(mode)
-  await git(['reset', modeFlag, ref, '--'], repository.path, 'reset')
+  const args = resetModeToArgs(mode, ref)
```

Comment:
I _think_ we should still have the trailing `--` so that Git provides a better error message than complaining about ambiguity.

Reply:
😂😂😂😂

Reply:
Not sure I understand. This is still the case for the non-Win32 shells:

https://github.com/desktop/desktop/blob/b8c18c3fae807f5e12ba04c513f973cf132da2a4/app/src/lib/git/reset.ts#L91-L94

Reply:
And in this case, the `--` is unnecessary because it doesn't provide any files.

Reply:
Ugh, I didn't realise this is only supported for mixed resets:

<img width="559" src="https://user-images.githubusercontent.com/359239/30887531-35fa72b2-a35f-11e7-91e4-491323568389.png">

Back to the drawing board...

Reply:
Nevermind, the workaround wasn't too bad 📝